File: TUF/String/SumOfNumbersInString.ts
function sumOfNumbersInString(str: string): number {
    let tempSum: string = "0";
    let sum: number = 0;
    for (let i: number = 0; i < str.length; i++) {
        let currChar: string = str.charAt(i);
        if(currChar >= "0" && currChar <= "9") {
            tempSum += currChar;
        }else{
            sum += Number(tempSum);
            tempSum = "0";
        }
    }
    return sum + Number.parseInt(tempSum);
}

let str: string = "1xyz23";
console.log(`The string: ${str}`);
console.log(`The sum: ${sumOfNumbersInString(str)}`);

export { };

File: TUF/String/SumOfNumbersInString.java
import java.util.Scanner;

public class SumOfNumbersInString {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the string: ");
        String str = input.next();
        input.close();
        System.out.println("The string: " + str);
        System.out.println("The sum: " + sumOfNumbersInString(str));
    }

    private static int sumOfNumbersInString(String str) {
        String tempSum = "0";
        int sum = 0;
        for (int i = 0; i < str.length(); i++) {
            char currChar = str.charAt(i);
            if (Character.isDigit(currChar)) {
                tempSum += currChar;
            } else {
                sum += Integer.parseInt(tempSum);
                tempSum = "0";
            }
        }
        return sum + Integer.parseInt(tempSum);
    }
}
